src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -320,6 +326,25 @@ public FormValidation doCheckHookUrl(@QueryParameter String value) {
 
         }
 
+        public FormValidation doReRegister() {
+            if (!manageHook) {
+                return FormValidation.error("Works only when jenkins manages hooks");
+            }
+
+            int triggered = 0;
+            for (AbstractProject<?,?> job : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
+                GitHubPushTrigger trigger = job.getTrigger(GitHubPushTrigger.class);
+                if (trigger!=null) {
+                    LOGGER.log(Level.FINE, "Calling registerHooks() for {0}", job.getFullName());
+                    trigger.registerHooks();
+                    triggered++;
+                }
+            }
+
+            LOGGER.log(Level.INFO, "Called registerHooks() for {0} jobs", triggered);
```

Comment:
I would recommend to have something human-readable. Jenkins users do not know what <code>registerHooks()</code> means